stdlib/unittest/runner.pyi

```diff
@@ -2,36 +2,48 @@ import sys
 import unittest.case
 import unittest.result
 import unittest.suite
-from _typeshed import Incomplete
 from collections.abc import Callable, Iterable
-from typing import TextIO
-from typing_extensions import TypeAlias
+from typing import Any, Generic, Protocol, TextIO, TypeVar, type_check_only
+from typing_extensions import Never, TypeAlias
 
-_ResultClassType: TypeAlias = Callable[[TextIO, bool, int], unittest.result.TestResult]
+# Build a Protocol from TextIO's methods.
+# `TextTestRunner.stream` defaults to `sys.stderr` which is typed as `TextIO`
+@type_check_only
+class _TextIOLike(TextIO, Protocol): ...  # type: ignore[misc] # pyright: ignore[reportGeneralTypeIssues]
```

Comment:
I'd just use the following, as that is what is actually used by the decorator:

```suggestion
@type_check_only
class _TextIOLike(Protocol):
    def write(self, s: str) -> Unused: ...
```

Reply:
Agree. Also, inheriting from both a concrete class and a Protocol isn't meaningful in the type system. We can type ignore it away, but that doesn't make it meaningful.